### PR TITLE
Advanced Type: Quiz

### PR DESCRIPTION
**Pull request by leogermond**
- adv subtypes: quiz
- lint 030

### CODE REVIEW

File: courses/fundamentals_of_ada/adv_030_basic_types.rst [hunk 1]
```diff
@@ -483,6 +579,44 @@ Shift/Rotate for User-Defined Types
...
+
+    with Interfaces; use Interfaces;
+
+    type T1 is new Unsigned_8;
+    V1 : T1 := 255;
+
+    type T2 is mod 256;
+    V2 : T2 := 255;
+
+Which statements are correct?
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_030_basic_types.rst:612]: Same as previous comment

File: courses/fundamentals_of_ada/adv_030_basic_types.rst [hunk 2]
```diff
@@ -635,3 +785,37 @@ Ada.Characters.Handling Sample Content
...
+Quiz
+------
+
+.. code:: Ada
+
+    type T1 is (NUL = 0, A, B, 'C');
+    type T2 is array (Positive range <>) of T1;
+    Obj : T2 := "CC" & A & NUL;
+
+Which of the following propositions are true?
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_030_basic_types.rst:798]: In the phrasing of this and the previous, the student can't tell if there should be only 1 answer, or 1 or more answers

**Review by frank-at-adacore — DISMISSED** [review]
General comment about the quizzes:
When doing these online, I use a webex/meet poll that I have stored in a file (so I don't have to populate the poll every time). Previous quizzes always had four possible answers, and one possible solution, so I could always use the same file.
As I'm not sure how much "advanced" we'll be doing online, not sure it's a big deal here, but it is something to consider

**Comment by leogermond** [comment]
Not sure I understand the webex file stuff, I guess it's possible to have several of them pre-saved? Don't hesitate to share this kind of resources in the future, I'm interested.

File: courses/fundamentals_of_ada/adv_030_basic_types.rst [hunk 3]
```diff
@@ -205,6 +205,84 @@ Attributes Reflect the Underlying Type
...
+------
+Quiz
+------
+
+.. code:: Ada
+
+    type T1 is new Integer range 0 .. 10;
+    subtype T2 is T1 range 1 .. 9;
+
+What is the result of :ada:`T2'Base`?
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_030_basic_types.rst:217]: Maybe type then? Range in that context would be understood as the Ada reserved word, but that's not the question.

File: courses/fundamentals_of_ada/adv_030_basic_types.rst [hunk 4]
```diff
@@ -332,6 +410,24 @@ Using Equality for Floating Point Types
 
    - Comparison within tolerance (:math:`+\varepsilon` / :math:`-\varepsilon`)
 
+------
+Quiz
```
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_030_basic_types.rst:414]: Right, fixed it

File: courses/fundamentals_of_ada/adv_030_basic_types.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (frank-at-adacore) [reply courses/fundamentals_of_ada/adv_030_basic_types.rst]: 1) Range of T2 is 1..9, range of T1 is 0..10 - that's what I mean
2) 'Base gives you the range of the _number of bits_ used, so answer A is actually correct (try writing a simple program with    "ada.text_io.put_line(t2'base'first'image);"

**Comment by frank-at-adacore** [comment]
> Not sure I understand the webex file stuff, I guess it's possible to have several of them pre-saved? Don't hesitate to share this kind of resources in the future, I'm interested.

Good idea. I have a Webex file, I'll create a Google Meet version and push them to the repository

File: courses/fundamentals_of_ada/adv_030_basic_types.rst [hunk 5]
```diff
@@ -214,7 +214,7 @@ Quiz
     type T1 is new Integer range 0 .. 10;
     subtype T2 is T1 range 1 .. 9;
 
-What is the result of :ada:`T2'Base`?
+What is the :ada:`type` of :ada:`T2'Base`?
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_030_basic_types.rst:217]: This quiz looks duplicated? (First occurrence has wrong answer)

File: courses/fundamentals_of_ada/adv_030_basic_types.rst [hunk 6]
```diff
@@ -588,7 +588,7 @@ Quiz
     type T is mod 256;
     V : T := 255;
 
-Which statements are correct?
+Which statements are legal?
```
Comment (frank-at-adacore) [inline courses/fundamentals_of_ada/adv_030_basic_types.rst:591]: should be "statement(s) is(are)"

**Review by frank-at-adacore — CHANGES_REQUESTED** [review]

File: courses/fundamentals_of_ada/adv_030_basic_types.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (leogermond) [reply courses/fundamentals_of_ada/adv_030_basic_types.rst]: Ah I see, let's remove that question then, it is in the chapter about base types

**Review by frank-at-adacore — APPROVED** [review]

**Pull request merged**